tests/phpunit/tests/pluggable.php

```diff
@@ -324,4 +325,24 @@ public function get_pluggable_function_signatures() {
 		return $signatures;
 	}
 
+	/**
+	 * @ticket 28020
+	 */
+	public function test_get_user_by_should_return_same_instance_as_wp_get_current_user() {
```

Comment:
If it doesn't already exist in capabilities or elsewhere, it would be dandy to test that `switch_to_blog()` invalidates the caps of the current user when you `get_user_by()`, etc.

Reply:
I changed this to assertSame as the objects are passed by reference so no need to make the data change to ensure the objects are identical. Makes the tests slightly speedier.

Reply:
To confirm this test does reveal the bug and show the PR does fix it, I removed the src changes and ran the test. Results:

```
1) Tests_Pluggable::test_get_user_by_should_return_same_instance_as_wp_get_current_user
Failed asserting that two variables reference the same object.
```

Reapplying the src changes and rerunning the tests results in this test passing.

✅ Confirms this test is validate

Reply:
Validate if this test does reveal the bug by:

- Comment out the src changes
- Run: `npm run test:php -- --group pluggable`
- Result:

```
1) Tests_Pluggable::test_get_user_by_should_return_same_instance_as_wp_get_current_user
Failed asserting that two variables reference the same object.

/var/www/tests/phpunit/tests/pluggable.php:340
```

✅ Yes, this test does reveal the bug before the fix

- Uncomment the src changes
- Rerun the tests
- Tests all pass

✅ Yes, this test does validate the src changes to fix the bug.